grammars/gfm.cson

```diff
@@ -867,6 +867,30 @@
         'name': 'markup.underline.link.gfm'
   }
   {
+    'begin': '^\\s*(- \\[ \\])(\\s*)(.*)'
+    'end': '$'
+    'name': 'task.unordered.list.gfm'
+    'captures':
+      '1':
+        'name': 'punctuation.gfm'
+      '2':
+        'name': 'space.gfm'
+      '3':
+        'name': 'text.gfm'
+  }
+  {
+    'begin': '^\\s*(- \\[x\\])(\\s*)(.*)'
```

Comment:
I'm not sure you want the `(.*)` at the end of this pattern, if you instead use the `$self` technique that I commented below, it will support parsing bold, italic, etc. in between the `- [x]` and the end of the line.